app/src/main/java/com/example/benchproject/data/movie/details/repo/MovieDetailsDataRepository.kt

```diff
@@ -0,0 +1,31 @@
+package com.example.benchproject.data.movie.details.repo
+
+import com.example.benchproject.data.movie.details.entity.MovieDetailsRemote
+import com.example.benchproject.data.movies.common.api.MoviesApi
+import com.example.benchproject.data.movies.common.constants.MoviesConstants.BASE_URL_FOR_IMAGE
+import com.example.benchproject.domain.movie.details.entity.GenresModel
+import com.example.benchproject.domain.movie.details.entity.MovieDetailsModel
+import com.example.benchproject.domain.movie.details.repo.MovieDetailsRepository
+import javax.inject.Inject
+
+class MovieDetailsDataRepository @Inject constructor(
+    private val apiService: MoviesApi
+) : MovieDetailsRepository {
+    override suspend fun getMovieDetails(movieId: Int) = runCatching {
+        apiService.getMovieDetails(movieId).mapToMovieDetailsModelType()
```

Comment:
You should also unit test that api was called, for such cases use `verify { some mock call }`

Reply:
I prepared a special test case for this :D